src/strategy/filesystem.rs

```diff
@@ -0,0 +1,180 @@
+//! Strategy for local filesystem-based updates.
+
+use anyhow::{Context, Error, Result};
+use fn_error_context::context;
+use futures::future;
+use futures::prelude::*;
+use log::trace;
+use serde::{Deserialize, Serialize};
+use std::os::unix::fs::{MetadataExt, PermissionsExt};
+use std::pin::Pin;
+use std::time::{SystemTime, UNIX_EPOCH};
+use tokio::fs as tokio_fs;
+
+/// Struct to parse finalization marker file's JSON content into.
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct FinalizationMarker {
+    /// Unix timestamp of expiry time.
+    not_after: Option<u64>,
+}
+
+/// Strategy for immediate updates.
+#[derive(Clone, Debug, Default, Serialize)]
+pub(crate) struct StrategyFilesystem {}
+
+impl StrategyFilesystem {
+    /// Strategy label/name.
+    pub const LABEL: &'static str = "filesystem";
```

Comment:
I'd try to find something more descriptive. Ideally the name would convey at least some of following key concepts:
 * it's a local marker (i.e. it has no fleet-wide coordination)
 * it defines a time-window
 * it can persist across reboots
 * it's a level trigger (i.e. it can hold into a state, and can be queried over time)
 * it's baked by a regular file

Reply:
How about `marker_file` strategy.  Conveys first point and last point and pretty simple to understand :)